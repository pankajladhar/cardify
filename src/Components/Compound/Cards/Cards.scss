.Cards {
  .Card {
    margin-bottom: 20px;
  }
}