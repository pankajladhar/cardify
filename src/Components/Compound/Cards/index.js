import React, { Component, Fragment } from 'react';
import Loader from './../../Base/Loader';
import Card from './../../Base/Card';
import { getSavedCards } from './../../../API';

import './Cards.css';

class Cards extends Component {
    constructor(props) {
        super(props);
        this.state = {
            cards: []
        };
        this.renderLoader = this.renderLoader.bind(this);
        this.renderCards = this.renderCards.bind(this);
    }

    componentDidMount() {
        this.setState({
            cards: getSavedCards()
        });
    }

    renderCards() {
        return (
            <div className="Cards">
                {
                    this.state.cards.map(card =>
                        <Card key={card.id} {...card}/>
                    )
                }
            </div>
        )
    }

    renderLoader() {
        return (
            <Loader />
        )
    }

    render() {
        return (
            <Fragment>
                {
                    this.state.cards.length > 0 ?
                        this.renderCards()
                        : this.renderLoader()
                }
            </Fragment>
        )
    }
}

export default Cards