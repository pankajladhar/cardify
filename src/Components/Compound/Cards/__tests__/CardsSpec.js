import React from 'react';
import renderer from 'react-test-renderer';
import Cards from './../index';

describe('Cards', () => {
    it('should render Loader screen when Cards state is empty', () => {
        let component = renderer.create(
            <Cards />
        );
        let instance = component.getInstance();
        instance.state.cards = [];
        expect(component.toJSON()).toMatchSnapshot(); 
    });

    it('should render correctly', () => {
        let tree = renderer.create(
            <Cards />
        ).toJSON();

        expect(tree).toMatchSnapshot()
    })
})