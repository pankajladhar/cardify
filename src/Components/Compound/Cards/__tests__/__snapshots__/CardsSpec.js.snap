// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should render Loader screen when Cards state is empty 1`] = `
<div
  className="Cards"
>
  <div
    className="Card"
  >
    HDFC
     
    HDFC Card
  </div>
  <div
    className="Card"
  >
    CITI Bank
     
    CITI Bank Card
  </div>
  <div
    className="Card"
  >
    TATA SBI
     
    TATA SBI Card
  </div>
</div>
`;

exports[`Cards should render correctly 1`] = `
<div
  className="Cards"
>
  <div
    className="Card"
  >
    HDFC
     
    HDFC Card
  </div>
  <div
    className="Card"
  >
    CITI Bank
     
    CITI Bank Card
  </div>
  <div
    className="Card"
  >
    TATA SBI
     
    TATA SBI Card
  </div>
</div>
`;
